purchases/src/main/kotlin/com/revenuecat/purchases/SyncPendingTransactionsHelper.kt

```diff
@@ -0,0 +1,114 @@
+package com.revenuecat.purchases
+
+import com.revenuecat.purchases.common.AppConfig
+import com.revenuecat.purchases.common.BillingAbstract
+import com.revenuecat.purchases.common.Dispatcher
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.caching.DeviceCache
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.models.StoreTransaction
+import com.revenuecat.purchases.strings.PurchaseStrings
+import com.revenuecat.purchases.strings.RestoreStrings
+import com.revenuecat.purchases.utils.Result
+
+internal class SyncPendingTransactionsHelper(
+    private val appConfig: AppConfig,
+    private val deviceCache: DeviceCache,
+    private val billing: BillingAbstract,
+    private val dispatcher: Dispatcher,
+    private val identityManager: IdentityManager,
+    private val postTransactionWithProductDetailsHelper: PostTransactionWithProductDetailsHelper,
+) {
+
+    fun syncPendingPurchaseQueue(
+        allowSharingPlayStoreAccount: Boolean,
+        onError: ((PurchasesError) -> Unit)? = null,
+        onSuccess: ((CustomerInfo?) -> Unit)? = null,
```

Comment:
The callback didn't exist before. Right now it's unused, but my plan is to use it once we want to sync purchases before fetching customer info.